File: javaTask/Editor.java
package javaTask;

public class Editor {
    public String filePath;

    public Editor(String filePath) {
        this.filePath = filePath;
    }
    
    private final StringBuilder textBuilder = new StringBuilder();

    private int position = 0;

    public String getText() {
        return textBuilder.toString();
    } 

    public void leftArrow() {
        position = Math.max(0, position - 1);
    }

    public void rightArrow() {
        position = Math.min(position + 1, getText().length());
    }
    
    public void type(String text) {
        textBuilder.insert(position, text);
        position += text.length();
    }
    
    public void pressEnter() {
        type("\n");
    }

    public void pressBackspace() {
        if (position != 0) {
            textBuilder.delete(position - 1, position);
            position--;
        }
    }

    private boolean isOpened = true;

    public void close() {
        isOpened = false;
    }

    public boolean isOpened() {
        return isOpened;
    }
}

File: kotlinTask/Editor.kt
package kotlinTask;

class Editor(val filePath: String) {
    private var textBuilder = StringBuilder()

    private var position: Int = 0

    fun getText() = textBuilder.toString()

    fun leftArrow() {
        position = maxOf(0, position - 1)
    }

    fun rightArrow() {
        position = minOf(position + 1, getText().length)
    }

    fun type(text: String) {
        textBuilder.insert(position, text)
        position += text.length
    }

    fun pressEnter() = type("\n")

    fun pressBackspace() {
        if (position != 0) {
            textBuilder.delete(position - 1, position)
            position--;
        }
    }
    
    private var isOpened = true

    fun close() {
        isOpened = false
    }

    fun isOpened() = isOpened
}